### PR TITLE
Use http connection pool for the clients

### PR DESCRIPTION
**Pull request by danpersa**

### CODE REVIEW

File: src/main/java/org/zalando/spearheads/innkeeper/utils/OAuthServiceClient.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package org.zalando.spearheads.innkeeper.utils;
```
Comment (mo-gr) [inline src/main/java/org/zalando/spearheads/innkeeper/utils/OAuthServiceClient.java]: `.java` ?

File: src/main/scala/org/zalando/spearheads/innkeeper/api/JsonDtos.scala [hunk 2]
```diff
@@ -20,9 +20,9 @@ object RouteName {
   val validRouteNamePattern = "[a-zA-Z][a-zA-Z0-9_]*"
 }
 
-case class TeamName(name: String) extends AnyRef
+case class TeamName(name: String) extends AnyVal
 
-case class UserName(name: String) extends AnyRef
+case class UserName(name: String) extends AnyVal
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/api/JsonDtos.scala]: What is the difference?

File: src/main/scala/org/zalando/spearheads/innkeeper/utils/HttpClient.scala [hunk 3]
```diff
@@ -67,3 +78,14 @@ class AkkaHttpClient @Inject() (
     }
   }
 }
+
+private object ConnectionPoolFactory {
+
+  def apply(uri: Uri)(implicit actorSystem: ActorSystem, materializer: ActorMaterializer): Flow[(HttpRequest, Int), (Try[HttpResponse], Int), HostConnectionPool] = {
+
+    uri.scheme match {
+      case "shttp" | "https" => Http().cachedHostConnectionPoolTls[Int](uri.authority.host.address())
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/utils/HttpClient.scala]: shttp?

**Comment by mo-gr** [comment]
I don't see anything wrong with this :+1:

File: src/main/java/org/zalando/spearheads/innkeeper/utils/OAuthServiceClient.java (hunk 1 above)
Comment (danpersa) [inline src/main/java/org/zalando/spearheads/innkeeper/utils/OAuthServiceClient.java]: it seems that you can't define java annotations in scala... and guice need java annotations :(

File: src/main/scala/org/zalando/spearheads/innkeeper/utils/HttpClient.scala (hunk 3 above)
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/utils/HttpClient.scala]: weird, but that's how the scheme is stored in akka http... maybe they have a bug...

File: src/main/scala/org/zalando/spearheads/innkeeper/api/JsonDtos.scala (hunk 2 above)
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/api/JsonDtos.scala]: http://docs.scala-lang.org/overviews/core/value-classes.html

**Pull request merged**